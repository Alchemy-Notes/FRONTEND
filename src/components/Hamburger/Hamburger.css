header {
  height: 50px;
  background-color: #ffb1a0;
  margin: 0;
}

.menubtn {
  height: 40px;
  border: none;
  background-color: #ffb1a0;
}

.menubtn > img {
  height: 40px;
}
