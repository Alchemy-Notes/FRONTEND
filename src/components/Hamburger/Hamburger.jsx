import { Link } from 'react-router-dom';

export default function Hamburger() {
  return (
    <>
      {/* className={styles.hamburger} */}
      <header>
        <Link to="/">Home</Link>
        <Link to="/login"></Link>
        <Link to="/about">About Us</Link>
        <Link to="/notes">Notes</Link>
        <Link to="/register">Log Out</Link>
        {/* -- this can redirect to /register and log someone out  */}
      </header>
    </>
  );
}
