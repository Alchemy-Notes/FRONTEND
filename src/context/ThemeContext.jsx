import { createContext, useContext, useState } from 'react';

const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState(false);

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

const useTheme = () => {
  const context = useContext(ThemeContext);

  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeContext Provider');
  }

  return context;
};

export { ThemeProvider, useTheme };
