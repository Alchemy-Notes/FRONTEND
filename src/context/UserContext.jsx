import { createContext, useContext, useEffect, useState } from 'react';
import { getMe } from '../services/auth';
import { getUserNotes } from '../services/notes';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [notes, setNotes] = useState(null);
  const [tree, setTree] = useState(null);

  useEffect(() => {
    async function get() {
      const result = await getMe();
      setUser(result);
    }
    get();
  }, []);

  useEffect(() => {
    async function getNotes() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) getNotes();
  }, [user]);

  return user && notes ? (
    <UserContext.Provider
      value={{ user, setUser, notes, setNotes, tree, setTree }}
    >
      {children}
    </UserContext.Provider>
  ) : (
    <></>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserContext Provider');
  }

  return context;
};

export { UserProvider, useUser };
