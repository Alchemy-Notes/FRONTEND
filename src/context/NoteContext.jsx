import { createContext, useContext, useEffect, useState } from 'react';
import { getUserNotes } from '../services/notes';
import { useUser } from './UserContext';

const NoteContext = createContext();

const NoteProvider = ({ children }) => {
  const [notes, setNotes] = useState([]);
  const { user } = useUser();

  useEffect(() => {
    async function get() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) get();
  }, []);

  return (
    <NoteContext.Provider value={{ notes, setNotes }}>
      {children}
    </NoteContext.Provider>
  );
};

const useNotes = () => {
  const context = useContext(NoteContext);

  if (context === undefined) {
    throw new Error('useNotes must be used within a NoteContext Provider');
  }

  return context;
};

export { NoteProvider, useNotes };
