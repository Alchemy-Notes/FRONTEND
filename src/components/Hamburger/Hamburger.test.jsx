import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/';
import { UserProvider } from '../../context/UserContext';
import Hamburger from './Hamburger';

it('should render our button to log into github', () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <Hamburger />
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
