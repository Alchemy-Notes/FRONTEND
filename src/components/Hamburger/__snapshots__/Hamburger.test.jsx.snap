// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our Hamburger component 1`] = `
<div>
  <header>
    <nav>
      <div>
        <div
          class="bm-overlay"
          style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
        />
        <div>
          <div
            class="bm-burger-button"
            style="z-index: 1000; position: fixed; width: 36px; height: 30px; left: 36px; top: 20px;"
          >
            <button
              id="react-burger-menu-btn"
              style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
            >
              Open Menu
            </button>
            <span>
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1; background: rgb(255, 255, 255);"
              />
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1; background: rgb(255, 255, 255);"
              />
              <span
                class="bm-burger-bars"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1; background: rgb(255, 255, 255);"
              />
            </span>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="bm-menu-wrap"
          id=""
          style="position: fixed; z-index: 1100; width: 300px; height: auto; transform: translate3d(-100%, 0, 0); transition: all 0.5s; top: 69px;"
        >
          <div
            class="bm-menu"
            style="height: 100%; box-sizing: border-box; overflow: auto; background: rgb(255, 221, 214); border: 3px solid #ffb1a0; border-radius: 20px; padding: 2px; font-size: 1.15em;"
          >
            <nav
              class="bm-item-list"
              style="height: 100%; color: rgb(184, 183, 173); padding: 0.8em;"
            >
              <ul
                class="bm-item"
                style="display: inline-block;"
                tabindex="-1"
              >
                <li>
                  <a
                    class="menu-item"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <hr />
                <li>
                  <a
                    class="menu-item"
                    href="/notes"
                  >
                    Notes
                  </a>
                </li>
                <hr />
                <li>
                  <a
                    class="menu-item"
                    href="/notes/new"
                  >
                    New Note
                  </a>
                </li>
                <hr />
                <li>
                  <a
                    class="menu-item"
                    href="/about"
                  >
                    About Us
                  </a>
                </li>
                <hr />
                <li>
                  <a
                    class="menu-item"
                    href="/login"
                  >
                    Log Out
                  </a>
                </li>
              </ul>
            </nav>
          </div>
          <div>
            <div
              class="bm-cross-button"
              style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
            >
              <button
                id="react-burger-cross-btn"
                style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                tabindex="-1"
              >
                Close Menu
              </button>
              <span
                style="position: absolute; top: 6px; right: 14px;"
              >
                <span
                  class="bm-cross"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg); background: rgb(189, 195, 199);"
                />
                <span
                  class="bm-cross"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg); background: rgb(189, 195, 199);"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
